.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,74 @@
+name: Integration tests
+on:
+  push:
+    paths-ignore:
+      - 'doc/**'
+  pull_request:
+    paths-ignore:
+      - 'doc/**'
+
+jobs:
+  run-tests:
+    name: Intergration tests
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        deployment-type: [ 'helm', 'operator', 'kubectl' ]
+        client-language: [ 'java', 'nodejs' ]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - uses: engineerd/setup-kind@v0.5.0
+        with:
+          version: "v0.9.0"
+
+      - name: Deploy Hazelcast cluster (Helm)
+        if: startsWith(matrix.deployment-type, 'helm')
+        run: |-
+          helm repo add hazelcast https://hazelcast-charts.s3.amazonaws.com/
+          helm repo update
+          helm install hz-hazelcast hazelcast/hazelcast
+
+      - name: Deploy Hazelcast cluster (Hazelcast Operator)
+        if: startsWith(matrix.deployment-type, 'operator')
+        run: |-
+          kubectl apply -f https://raw.githubusercontent.com/hazelcast/hazelcast-operator/master/hazelcast-operator/operator-rbac.yaml
+          kubectl apply -f https://raw.githubusercontent.com/hazelcast/hazelcast-operator/master/hazelcast-operator/hazelcast-rbac.yaml
+          kubectl apply -f https://raw.githubusercontent.com/hazelcast/hazelcast-operator/master/hazelcast-operator/hazelcastcluster.crd.yaml
+          kubectl --validate=false apply -f https://raw.githubusercontent.com/hazelcast/hazelcast-operator/master/hazelcast-operator/operator.yaml
+          kubectl apply -f https://raw.githubusercontent.com/hazelcast/hazelcast-operator/master/hazelcast-operator/hazelcast.yaml
+          sleep 30
+
+      - name: Deploy Hazelcast cluster (Kubectl)
+        if: startsWith(matrix.deployment-type, 'kubectl')
+        env:
+          HAZELCAST_VERSION: 4.1
+        run: |-
+          kubectl apply -f https://raw.githubusercontent.com/hazelcast/hazelcast-kubernetes/master/rbac.yaml
+          kubectl run hz-hazelcast-0 --image=hazelcast/hazelcast:$HAZELCAST_VERSION -l "role=hazelcast"
+          kubectl run hz-hazelcast-1 --image=hazelcast/hazelcast:$HAZELCAST_VERSION -l "role=hazelcast"
+          kubectl run hz-hazelcast-2 --image=hazelcast/hazelcast:$HAZELCAST_VERSION -l "role=hazelcast"
+          kubectl create service clusterip hz-hazelcast --tcp=5701 -o yaml --dry-run=client | kubectl set selector --local -f - "role=hazelcast" -o yaml | kubectl create -f -
+
+      - name: Wait for Hazelcast cluster
+        run: |-
+          kubectl wait --for=condition=ready pod/hz-hazelcast-0 --timeout=150s
+          kubectl wait --for=condition=ready pod/hz-hazelcast-1 --timeout=150s
+          kubectl wait --for=condition=ready pod/hz-hazelcast-2 --timeout=150s
+
+      - name: Build client application
+        run: |-
+          docker build -t hazelcast-client:test ${{ matrix.client-language }}
```

Comment:
Good idea, but I guess we can keep it as it is since building the client app image takes only a small fraction of the total pipeline execution time.

![image](https://user-images.githubusercontent.com/2834997/102084623-bb260b00-3e15-11eb-911d-d8ab48030507.png)